geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/Ping.java

```diff
@@ -60,6 +78,29 @@ public void cmdExecute(final Message clientMessage, final ServerConnection serve
     }
   }
 
+  /**
+   * Process a ping request that was sent to the wrong server
+   */
+  protected void pingCorrectServer(Message clientMessage, DistributedMember targetServer,
+      ServerConnection serverConnection)
+      throws IOException {
+    if (logger.isDebugEnabled()) {
+      logger.debug("Received a Ping request from {} intended for {}.  Forwarding the ping...");
+    }
+    if (!serverConnection.getCache().getDistributionManager().isCurrentMember(targetServer)) {
+      logger.warn("Unable to ping non-member {} for client {}", targetServer,
+          serverConnection.getProxyID());
+      writeErrorResponse(clientMessage, MessageType.PING, serverConnection);
+      serverConnection.setAsTrue(RESPONDED);
+    } else {
+      // send a ping message to the server. This is a one-way message that doesn't send a reply
+      final DistributedPingMessage distributedPingMessage =
+          new DistributedPingMessage(targetServer, serverConnection.getProxyID());
```

Comment:
Does this need to take into account the version of the member that is being pinged? Could we be pinging a member with an older version and it will fail to deserialize this message? Or are we assuming all of the receivers will be upgraded before we start getting pings that need to be forwarded to the correct server.

Reply:
In our case its ok if the ping is not working while the upgrade of all members is not finished

Reply:
I don't think Ping needs to check the version of the recipient of the DistributedPingMessage.  Only  new clients will include the new targetServer message part, and all servers should be upgraded before new clients come on-line.